src/gui/settingsdialog.cpp

```diff
@@ -175,6 +178,12 @@ void SettingsDialog::changeEvent(QEvent *e)
     QDialog::changeEvent(e);
 }
 
+void SettingsDialog::setVisible(bool visible)
```

Comment:
Same comments than before on the other dialog: event handler would be better, leaving that completely out of the dialog and using an event filter instead would be best.